src/Pages/Solver.tsx

```diff
@@ -1,19 +1,213 @@
+import React from 'react';
 import './../App.css';
+import './../Util/dcr.js';
+import {parser} from './../Util/dcr_parser.js';
+import {dynamicTable} from './../Util/dynamic_table.js';
 
-const Home = () => {
-  return (
-    <div className="App">
-    <h1 style={{ textAlign: 'center' }}>JavaScript DCR Graphs Engine</h1>
+import { Header } from '../Components/HeaderSolver';
+import { HelpSolver } from '../Components/HelpSolver';
+import { Footer } from '../Components/Footer';
 
-    <p style={{ textAlign: 'center' }}>
-      <a href="creator.html"><button id="btn-front-page">Creator</button></a>
-      <a href="solver.html"><button id="btn-front-page">Solver</button></a>
-      <button id="btn">Test Export</button>
-    </p>
+import Grid from '@mui/material/Grid';
+import Paper from '@mui/material/Paper';
+import { Alert, List, ListItemButton, ListItemIcon, ListItemText, TextField } from '@mui/material';
+import HelpIcon from '@mui/icons-material/Help';
+import UploadFileIcon from '@mui/icons-material/UploadFile';
+import HintIcon from '@mui/icons-material/Lightbulb';
+import NextIcon from '@mui/icons-material/PlayArrow';
 
-    <script src="js/Control/exercise_logic.js"></script>
-    </div>
-  );
+interface Symbol {
+  symbol: string;
+  activity: string;
 }
 
-export default Home;
+class ExerciseSolver {
+  question: string;
+  symbols: Symbol[];
+  constructor() {
+    this.question = "";
+    this.symbols = [{ symbol: "", activity: "" }];
+  }
+}
+
+// Main page of the application
+class Solver extends React.Component<any, any> {
+  constructor(props: any) {
+    super(props);
+    this.state = {
+      exercise: new ExerciseSolver(),
+      parseError: "",
+    };
+  }
+
+  // Handles changes to the form and updates the state
+  handleChange(i: number, e: any) {
+    let newExercise = this.state.exercise;
+    switch (e.target.id) {
+      case "question":
+        newExercise.question = e.target.value;
+        break;
+      case "sym":
+        newExercise.symbols[i].symbol = e.target.value;
+        break;
+      case "act":
+        newExercise.symbols[i].activity = e.target.value;
+        break;
+      case "scn":
+        newExercise.scenarios[i].scenario = e.target.value;
+        break;
+      case "hnt":
+        newExercise.scenarios[i].hint = e.target.value;
+        break;
+    }
+    this.setState({ exercise: newExercise });
+  }
+
+  parseSolution(e: any) {
+    try {
+      let graph = parser.parse(e.target.value);
+      this.setState({parseError: ""})
+    } catch (err: any) {
+      this.setState({parseError: (err.message + "</br>" + JSON.stringify(err.location))})
+    }
+
+  }
+
+  // Adds a new symbol/activity mapping to the exercise
+  addSymbolFields() {
```

Comment:
Not sure this is needed, this will alter the exercise? Solver should be read-only

Reply:
This and below is the functionality that might be needed to read in the data from the json file (the buttons are gone) - so they are present as functions. Agree that Solver should be read-only. My thought is that reading in data could be similar in the Creator and Solver. If another solution works, fine to choose that.

Reply:
Keeping this for now, but have included comments in the code.